### PR TITLE
SQLStore: extend `user.SearchUsers` method

### PR DESCRIPTION
**Pull request by markelog**
* Allow `user.SearchUsers` to search users based on their auth type

* don't use shorten name for models module

### CODE REVIEW

File: pkg/services/sqlstore/user.go [hunk 1]
```diff
@@ -447,6 +447,17 @@ func SearchUsers(query *m.SearchUsersQuery) error {
 		whereParams = append(whereParams, queryWithWildcards, queryWithWildcards, queryWithWildcards)
 	}
 
+	if query.AuthModule != "" {
```
Comment (markelog) [inline pkg/services/sqlstore/user.go]: Here is the only real change in this file

File: pkg/services/sqlstore/user.go [hunk 2]
```diff
@@ -447,6 +447,17 @@ func SearchUsers(query *m.SearchUsersQuery) error {
...
+		whereConditions = append(
+			whereConditions, fmt.Sprintf(
+				`id IN (SELECT "user_id"
+				FROM "user_auth"
+				WHERE auth_module='%s')`,
```
Comment (torkelo) [inline pkg/services/sqlstore/user.go]: I think whereParams should be used, using string interpolation for a filter param opens up SQL injection possibilities, maybe not in this case as auth module is a known quantity but still think there is no reason to not add AuthModule to whereParams

Comment (markelog) [reply pkg/services/sqlstore/user.go]: Good point!

**Review by bergquist — APPROVED** [review]
LGTM

Nice to see removed alias for models import :)

**Review by xlson — APPROVED** [review]
LGTM!

**Pull request merged**